pkg/autoscaler/http_scrape_client.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+type httpScrapeClient struct {
+	httpClient *http.Client
+}
+
+func newHTTPScrapeClient(httpClient *http.Client) (*httpScrapeClient, error) {
+	if httpClient == nil {
+		return nil, errors.New("HTTP client must not be nil")
+	}
+
+	return &httpScrapeClient{
+		httpClient: httpClient,
+	}, nil
+}
+
+func (c *httpScrapeClient) Scrape(url string) (*Stat, error) {
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
+		return nil, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
+	}
+
+	return extractData(resp.Body)
+}
+
+func extractData(body io.Reader) (*Stat, error) {
+	var parser expfmt.TextParser
+	metricFamilies, err := parser.TextToMetricFamilies(body)
+	if err != nil {
+		return nil, fmt.Errorf("reading text format failed: %v", err)
+	}
+
+	now := time.Now()
```

Comment:
inline this.

Reply:
`time.Now()` is not addressable.

Reply:
Well, question begs: why is it a pointer in `Stat`? :-)